gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/GitFlowGraphMonitor.java

```diff
@@ -120,15 +119,19 @@ void processGitConfigChanges() throws GitAPIException, IOException {
     }
 
     List<DiffEntry> changes = this.gitRepo.getChanges();
-    Collections.sort(changes, (o1, o2) -> {
-      Integer o1Depth = (o1.getNewPath() != null) ? (new Path(o1.getNewPath())).depth() : (new Path(o1.getOldPath())).depth();
-      Integer o2Depth = (o2.getNewPath() != null) ? (new Path(o2.getNewPath())).depth() : (new Path(o2.getOldPath())).depth();
-      return o1Depth.compareTo(o2Depth);
-    });
+    Collections.sort(changes, new GitFlowgraphComparator());
     processGitConfigChangesHelper(changes);
     //Decrements the latch count. The countdown latch is initialized to 1. So after the first time the latch is decremented,
     // the following operation should be a no-op.
     this.initComplete.countDown();
   }
 
+  class GitFlowgraphComparator implements Comparator<DiffEntry> {
```

Comment:
It could be static

Reply:
This is good too. Another simple way would have been to create a static field in the class
```private static Comparator comparator = (Comparator<DiffEntry>) (o1, o2) -> {
    Integer o1Depth = (o1.getNewPath() != null) ? (new Path(o1.getNewPath())).depth() : (new Path(o1.getOldPath())).depth();
    Integer o2Depth = (o2.getNewPath() != null) ? (new Path(o2.getNewPath())).depth() : (new Path(o2.getOldPath())).depth();
    return o1Depth.compareTo(o2Depth);
  };```